src/app/feature/stack/stack.service.ts

```diff
@@ -0,0 +1,114 @@
+import { Injectable } from "@angular/core";
+import { ModalController } from "@ionic/angular";
+import { IStackConfig, StackComponent } from "./components/stack/stack.component";
+import { SyncServiceBase } from "src/app/shared/services/syncService.base";
+import { TemplateActionRegistry } from "src/app/shared/components/template/services/instance/template-action.registry";
+
+interface StackModal extends HTMLIonModalElement {}
+
+@Injectable({
+  providedIn: "root",
+})
+export class StackService extends SyncServiceBase {
+  openStacks: HTMLIonModalElement[] = [];
+
+  constructor(
+    private modalCtrl: ModalController,
+    private templateActionRegistry: TemplateActionRegistry
+  ) {
+    super("stack");
+    this.init();
+  }
+
+  private init() {
+    this.registerTemplateActionHandlers();
+  }
+
+  private registerTemplateActionHandlers() {
+    this.templateActionRegistry.register({
+      nav_stack: async ({ args, params }) => {
+        const [actionId] = args;
+        const childActions = {
+          open: async () => {
+            const { template, title, show_close_button } = params;
```

Comment:
nit(non-blocking): might be better to add default fallback values here instead of later filtering out the `undefined` values from the service code. E.g. 
```ts
 const { template, title, show_close_button=true } = params;
```